plugins/modules/net_tools/nmcli.py

```diff
@@ -1065,6 +1064,9 @@ def _compare_conn_params(self, conn_info, options):
                     current_value = [re.sub(r'^{\s*ip\s*=\s*([^, ]+),\s*nh\s*=\s*([^} ]+),\s*mt\s*=\s*([^} ]+)\s*}', r'\1 \2 \3',
                                      route) for route in current_value]
                     current_value = [re.sub(r'^{\s*ip\s*=\s*([^, ]+),\s*nh\s*=\s*([^} ]+)\s*}', r'\1 \2', route) for route in current_value]
+                if key == self.mac_setting:
+                    # MAC addresses are case insensitive, nmcli always reports them in uppercase
+                    value = str.upper(value)
             elif key in param_alias:
                 real_key = param_alias[key]
                 if real_key in conn_info:
```

Comment:
... after the line following this one: at that point you need to check whether `key == 'mac'`, and if it is, convert `current_value` to upper case.